libs/datamodel/core/src/transform/ast_to_dml/generator_loader.rs

```diff
@@ -45,6 +47,12 @@ impl GeneratorLoader {
             .map(|arg| (arg.name.name.as_str(), ValueValidator::new(&arg.value)))
             .collect();
 
+        if let Some(val) = args.get(ENGINE_TYPE_KEY) {
+            if let Err(err) = StringFromEnvVar::try_from(val.value) {
```

Comment:
Do we want to support this key with an `env(...)` function? (I don't know what's the current behavior). Otherwise I think we have another method to force a string literal.

Reply:
Asked around, nobody seems to really know. We never use these values in the engines, and while we can set this with an env var, it has a different system and doesn't use the schema for reading the value.

Now we force this to be a string, making `env` not allowed.

[`4fdb815` (#2549)](https://github.com/prisma/prisma-engines/pull/2549/commits/4fdb815e661c5d9408772e4532f82c85a81ab32f)